packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts

```diff
@@ -240,24 +240,23 @@ describe('plugin-dss', () => {
 
         expect(result).to.be.null;
       });
-
       it('fails with default timeout when mercury does not respond', async () => {
-        const {promise} = await testMakeRequest({
+        return testMakeRequest({
           method: 'lookupDetail',
           resource: '/lookup/orgid/userOrgId/identity/test id/detail',
           params: {id: 'test id'},
           bodyParams: {},
+        }).then(async ({promise}) => {
```

Comment:
> This change doesn't look right. Now if promise doesn't reject, the test will not fail and it should.

I agree. We can add a then block and fail it there. Will do it. 

>  Now if promise doesn't reject, the test will not fail and it should. Also the clock.tickAsync() is now no longer called before the check for the promise being rejected, it may be called before or after, which will probably make the test more flakey.

So, a little context for you here:

Earlier, the mocha tests were letting the Promises that were rejected go without any breakage. So, the code reached the assertion for checking isRejected. With new tooling and stricter checks, the test breaks as soon as the promise rejects because there's no catch mechanism. 

In the current code in beta,

1. We create a promise
2. We tick the clock to 6s and as soon as that happens, the promise rejects
3. When the promise rejects, we check the assertion for `isRejected`

In the newer code from next,

1. We create a promise
2. Write a catch block for When the promise would fail (The code won't execute the error block before tickAsync is executed)
3. We do `clock.tickAsync()` which makes the promise reject
4. When this happens, it goes into the catch block and checks if it has received the expected error message

So, to address your concern, we are not checking if the promise was rejected before `clock.tickAsync()` rather just added a catch mechanism for whenever it fails. Adding a then mechanism if it passes and failing the test would make this test whole I believe.

Reply:
Ah. Clever one. Thanks, Marcin. We can do this. And also see if we can follow a similar approach to the above-skipped test.

> Yes, agreed, that would be fine, but I don't see promise.then() being done in this PR.

About this, we were waiting for your signal, if my explanation was convincing to you so that we can make the change. Now that we go with the old approach and add just a line, we won't need it.

Reply:
PR to revert this: https://github.com/webex/webex-js-sdk/pull/3564

Reply:
PR merged and cherry-picked into this PR. Please check

Reply:
I see it's fixed for this test, but there are at least 4 other tests with the same issue in this file that are not fixed